File: output/qwen_2.5_72b_instruct/avatar/algo_based_translation/python/java/codeforces_474_B.java

import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;

public class codeforces_474_B {
    public static void main(String[] args) {
        List<Integer> L = new ArrayList<>();
        Scanner scanner = new Scanner(System.in);
        scanner.nextLine(); // Read a line of input (not used)
        int r = 1;
        for (String k : scanner.nextLine().split(" ")) {
            int repeat = Integer.parseInt(k);
            for (int i = 0; i < repeat; i++) {
                L.add(r);
            }
            r++;
        }
        scanner.nextLine(); // Read a line of input (not used)
        for (String j : scanner.nextLine().split(" ")) {
            int index = Integer.parseInt(j) - 1;
            System.out.println(L.get(index));
        }
        scanner.close();
    }
}

File: output/qwen_2.5_72b_instruct/avatar/algo_based_translation/java/python/codeforces_474_B.py
import sys

def fun(arr, q, n, m):
    res = 0
    i = 0
    j = n
    while i <= j:
        md = i + (j - i) // 2
        if arr[md] == q:
            return md
        elif arr[md] > q:
            res = md
            j = md - 1
        else:
            i = md + 1
    return res

def main():
    import sys
    import sys
    from sys import stdin
    input = stdin.read().split()
    n = int(input[0])
    arr = [0] * n
    arr[0] = int(input[1])
    index = 2
    for i in range(1, n):
        x = int(input[index])
        arr[i] = x + arr[i - 1]
        index += 1
    m = int(input[index])
    index += 1
    q = [0] * m
    for i in range(m):
        q[i] = int(input[index])
        index += 1
    for k in range(m):
        print(fun(arr, q[k], n - 1, m) + 1)

if __name__ == "__main__":
    main()
